src/core/server/plugins/plugins_service.ts

```diff
@@ -82,17 +93,20 @@ export class PluginsService implements CoreService<PluginsServiceSetup, PluginsS
 
     const config = await this.config$.pipe(first()).toPromise();
 
+    let contracts = new Map<PluginName, unknown>();
     if (!config.initialize || this.coreContext.env.isDevClusterMaster) {
       this.log.info('Plugin initialization disabled.');
-      return {
-        contracts: new Map(),
-        uiPlugins: this.pluginsSystem.uiPlugins(),
-      };
+    } else {
+      contracts = await this.pluginsSystem.setupPlugins(deps);
     }
 
+    const uiPlugins = this.pluginsSystem.uiPlugins();
     return {
-      contracts: await this.pluginsSystem.setupPlugins(deps),
-      uiPlugins: this.pluginsSystem.uiPlugins(),
+      contracts,
+      uiPlugins: {
+        ...uiPlugins,
+        config: this.generateUiPluginsConfigs(uiPlugins.public),
```

Comment:
optional: The name collision probability is minimal. But to simplify reading, couldn't you handle it separately?
```js
{ contracts, uiPlugins, uiConfig }
```

Reply:
Yea, wasn't sure which one was the best approach. Both works with me, I can change it.

Reply:
https://github.com/elastic/kibana/pull/50641/commits/a87b7b061145ad03d12bb2569d59080943722e92 separates configs from plugins